packages/@uppy/provider-views/src/View.js

```diff
@@ -92,6 +92,11 @@ export default class View {
       if (file.author.url) tagFile.meta.authorUrl = file.author.url
     }
 
+    // add relativePath similar to non-remote files: https://github.com/transloadit/uppy/pull/4486#issuecomment-1579203717
+    if (file.relDirPath != null) tagFile.meta.relativePath = file.relDirPath ? `${file.relDirPath}/${tagFile.name}` : tagFile.name
```

Comment:
we currently change paths in three places: in `addPath`, mutation in `recursivelyListAllFiles`, and here. Would it be possible to have it centralised?

Reply:
should be possible to set the paths in `recursivelyListAllFiles` - c.f. https://github.com/owncloud/uppy/pull/1/commits/9f111c5ce8318548be2d2e7f11ce6069c8c5c653

I'm not handling absolute paths, but in @mifi 's implementation the `absPath` is available in `recursivelyListAllFiles` too.

Reply:
I think we need to set paths in multiple places, because not all files are loaded by recursivelyListAllFiles